### DISCUSSION TITLE
Is there a way to turn the mininmal debian into a network-enabled one?

### DISCUSSION THREAD

**Discussion by Cyberman-tM** [johang/sd-card-images#10] [Q&A]
Specifically for the Banana Pi Zero.
The image boots fine, but it seems there is no way to connect to the internet to download updates - I assume this is because the needed firmware would be propietary? Or because it can't be automated to update.

Is there a way to manually add those missing components? None of the usual "standard" tools exist on the image. Or maybe I'm doing everything wrong, I'm not excluding that...

**Comment by johang** [comment]
It should work. Make sure you use a recent image with the fix for issue #8.

**Reply by Goude63** [reply]
(Banana Pi M2 Zero bullseye image downloaded today)
ifconfig is only reporting "lo" interface and /sys/class/net is also only listing "lo"
On first boot attempts, boot process would freeze if the ethernet cable was connected.
After booting once without ethernet connected, the system boots ok but no ethernet hardware is shown.

**Reply by johang** [reply]
Show output of dmesg. Also try Bookworm or Sid.

**Reply by jjaard** [reply]
Just tried on 2 Banana Pi M2 Zero's the [image](https://sd-card-images.johang.se/boards/banana_pi_m2_zero.html) (debian-bookworm-armhf-bei4ah.bin.gz and boot-banana_pi_m2_zero.bin.gz respectively) and it does not getting network at all.
using `ifconfig` only loop IP is available

For example using [ALL-H3-CC H2+ (Tritium)](https://sd-card-images.johang.se/boards/libretech_all_h3_cc_h2_plus.html) image has internet but does not boot fully (as this is not BPi M2Z image which is expected behaiviour I suppose, sorry for mistakes in logs as they were copied from photo):

  | Couldn't find partition muc o |   |  
  | Can't set block device |   |  
  | zimage: Bad magic! |   |  
  | SCRIPT FAILED: continuing..• |   |  
  | MMC: no card present |   |  
  | ** Bad device specification mn 0 ** |   |  
  | Couldn't find partition mnc 0:1 |   |  
  | MC: no card present |   |  
  | ** Bad device specification mnc 0 ** |   |  
  | Couldn't find partition mnc 0:1 |   |  
  | MMC: no card present |   |  
  | ** Bad device specification mmc 0 ** |   |  
  | Couldn't find partition muc 0:1 |   |  
  | MM: no card present |   |  
  | No EFI system partition |   |  
  | No EFI system partition |   |  
  | Failed to persist EFI variables |   |  
  | BootOrder not defined |   |  
  | EFI boot manager: Cannot load any image |   |  
  | MMC: no card present |   |  
  | ** Bad device specification mc 0 xx |   |  

MMC: no card present

** Bad device specification mmc 0 **

Couldn't find partition mnc 0

Can't set block device

MMC: no card present

** Bad device specification mnc 0 xx

Couldn't find partition mnc o

Found legacy kernel configuration

MMC: no card present

Couldn't find partition mmc 0:1

Device 0: unknown device

ethernet01c30000 Waiting for PHY auto negotiation to complete. done

BOOTP broadcast 1

De arming: no co Re nane: ustng Con80008.g

Using ethernet@1c30000 device

TFTP from server 192.168.8.1: our IP address is 192.168.8.11

Filename

'COABOBOB. ing'.

Load address: 0x42000000

Loading: T T TITTTITT

Retry count exceeded: starting again missing environment variable: preuid

Retrieving file: pelinux.cig/01-02-81-4a-34-e8-df

Using ethernet01c30000 device

IFT from server 192.168.8.1; our IP address is 192.168.8.11

Filename pxelinux.cfg/01-02-81-4a-34-eB-df' •

Load address: 0x43200000

Loading: TITITTTITT

Retry count exceeded: starting again

Retrieving file: pelinux.cfg/COABOBOB

Using ethernet@1c30000 device

TFT from server 192.168.8.1: our IP address is 192.168.8.11

Filename

'pxelinux.cfg COASOBOB'.

Load address: 0x43200000

**Reply by johang** [reply]
BPi M2 Zero doesn't have ethernet on board, so not surprised ifconfig only shows lo. What did you expect?

**Reply by jjaard** [reply]
Hmm, I got confused as they are named the same by Banana Pi - used neofetch to check on Armbian:
OS: Armbian (22.11.0-trunk) armv7l
Host: Banana Pi BPI-P2-Zero / M2-Zero
Kernel: 5.19.6-sunxi
Uptime: 16 days, 17 hours, 56 mins
Packages: 1419 (dpkg)
Shell: zsh 5.8
Terminal: /dev/pts/1
CPU: Allwinner sun8i Family (4) @ 1.008GHz
Memory: 239MiB / 490MiB

**Reply by jjaard** [reply]
![Снимок экрана 2023-11-07 в 06 19 01](https://github.com/johang/sd-card-images/assets/114291676/d8db8c55-5fd7-426d-9437-684194e9e67d)

**Reply by jjaard** [reply]
Would appreciate if image for BPi P2 would also be available or P2 and M2 could be combined.

**Comment by juergenschorb** [comment]
My Bananapi M2 Ultra will not boot with the bullseye image. I follow the follow the instructions on the web-site:

$ wget https://dl.sd-card-images.johang.se/boots/2024-01-01/boot-banana_pi_m2_ultra.bin.gz
$ wget https://dl.sd-card-images.johang.se/debians/2024-01-08/debian-bookworm-armhf-eiph9o.bin.gz
$ zcat boot-banana_pi_m2_ultra.bin.gz debian-bookworm-armhf-eiph9o.bin.gz > sd-card.img
$ dd if=sd-card.img of=/dev/sdXXX

What I am doing wrong?

**Comment by wundercat** [comment]
Before I tried booting the sdcard, I mounted its Linux FS  and created a $MOUNTPT/root/.ssh/authorized_keys file with a public key generated with ssh-keygen. 

My Orange Pi Zero LTS booted Johan's image and got on the wired Ethernet.    I found its address with nmap and logged in with ssh.
The first-boot had expanded the FS to fill the sdcard.   The network is managed by systemd-networkd and systemd-resolved.
I noted the MAC address and assigned it a fixed IP address in the DHCP server.    If that's not convenient you could assign a static IP in /etc/systemd/network/whatever.network before booting. 

```
apt update
apt install tzdata locales
dpkg-reconfigure tzdata locales   # if needed
```

I also needed these.    YMMV.
`apt install iproute2 iptables nmap netcat-openbsd netstat \
  bind9-host less file vim lsof strace usbutils iputils-ping rsyslog`
